app/src/main/java/fr/free/nrw/commons/Achievements.java

```diff
@@ -0,0 +1,104 @@
+package fr.free.nrw.commons;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.os.Environment;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.Toast;
+import android.widget.Toolbar;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Random;
+
+import fr.free.nrw.commons.theme.NavigationBaseActivity;
+
+public class Achievements extends AppCompatActivity {
+
+    private static final double badge_image_ratio_width = 0.4;
+    private static final double badge_image_ratio_height = 0.36;
+    private ImageView imageView;
```

Comment:
Use butterknife instead of initing these views manually. :)

https://github.com/commons-app/apps-android-commons/wiki/Code-style